x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValueSources.java

```diff
@@ -71,19 +78,53 @@ public static List<ValueSourceInfo> sources(
             try {
                 fieldData = ctx.getForField(fieldType, MappedFieldType.FielddataOperation.SEARCH);
             } catch (IllegalArgumentException e) {
-                if (asUnsupportedSource) {
-                    sources.add(
+                switch (elementType) {
+                    case BYTES_REF -> sources.add(
                         new ValueSourceInfo(
                             new UnsupportedValueSourceType(fieldType.typeName()),
                             new UnsupportedValueSource(null),
                             elementType,
                             ctx.getIndexReader()
                         )
                     );
-                    continue;
-                } else {
-                    throw e;
+
+                    case LONG, INT -> sources.add(
+                        new ValueSourceInfo(CoreValuesSourceType.NUMERIC, ValuesSource.Numeric.EMPTY, elementType, ctx.getIndexReader())
+                    );
+
+                    case BOOLEAN -> sources.add(
+                        new ValueSourceInfo(CoreValuesSourceType.BOOLEAN, ValuesSource.Numeric.EMPTY, elementType, ctx.getIndexReader())
+                    );
+
+                    case DOUBLE -> sources.add(new ValueSourceInfo(CoreValuesSourceType.NUMERIC, new ValuesSource.Numeric() {
+                        @Override
+                        public boolean isFloatingPoint() {
+                            return true;
+                        }
+
+                        @Override
+                        public SortedNumericDocValues longValues(LeafReaderContext context) {
+                            return DocValues.emptySortedNumeric();
+                        }
+
+                        @Override
+                        public SortedNumericDoubleValues doubleValues(LeafReaderContext context) throws IOException {
+                            return org.elasticsearch.index.fielddata.FieldData.emptySortedNumericDoubles();
+                        }
+
+                        @Override
+                        public SortedBinaryDocValues bytesValues(LeafReaderContext context) throws IOException {
+                            return org.elasticsearch.index.fielddata.FieldData.emptySortedBinary();
+                        }
+
+                    }, elementType, ctx.getIndexReader()));
+                    default -> throw e;
                 }
+                HeaderWarning.addWarning(
```

Comment:
What kind of warning is this? In logs, or in the response header?

Reply:
Asking because I am not sure what is the expectation here: we find a value that cannot be extracted/rendered and we do what? Log a message or add a warning in the response header (like we do for other similar operations)?

Reply:
This is response header, I guess it's what the user needs the most as long as Kibana shows it together with the result.
We could also add a log debug though.

Reply:
There is a `Warnings` class we have for this. How about making that class the main point to add warnings no matter if they come from exceptions (like the `registerException` method) or from "unsupported" behavior?

Reply:
Good point, I'll give it a try

Reply:
Yes, that's recommended since that would give the user the location (i.e. query part) which generates the `null` -- useful if there are more columns "affected". But getting the source in here might be tricky.

Reply:
It was a good idea indeed, but as @bpintea said getting the source here is practically impossible; also at classpath level, the Compute module does not depend on QL, do it doesn't see `Source` class at all.